### PR TITLE
Clamp the `focusTab` action to the number of available tabs

### LINKED ISSUE #9369: Focus the last tab when the param to `focus-tab` is greater than the number of tabs
From releas notes
```
wt can now run commands in any window that already exists with wt -w # (#8898)

    You can specify wt -w -1 to force the creation of a new window even if you've disabled it with the Windowing Behaviors

          Specifying a window that does not exist (wt -w 1048576) will result in a new window being created with that ID (!)
```
it's great! 

but to make it actually useful it should work with same pattern for tabs so i'd to be able to

`wt -w 1048576 focus-tab -t 3`

and expected result is to focus tab with index 3

or create it if it not exists 

so for example 

if i have only one tab in this window perhaps it could fallback to

`wt -w 1048576 ; ; ; focus-tab -t 3` 

?

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

When we perform a `focusTab` action, we currently do nothing if the parameter was greater than the number of tabs. This PR changes that behavior. Now, `focus-tab -t 999999` will always focus the last tab, instead of silently doing nothing. 

## PR Checklist
* [x] Closes #9369 
* [x] I work here
* [x] Tests added/passed
* [n/a] Requires documentation to be updated

## Validation Steps Performed
* [x] ran tests
* [x] validated commandline manually

### CODE REVIEW

**Comment by DHowett** [comment]
Hmmmmmm.

If we're going down this road... For consistency's sake, I need ctrl+alt+9 to focus the last tab. That's how firefox, chromium, chrome, edge, *legacy edge*, and a bunch of other tabbed interfaces do it.

**Comment by DHowett** [comment]
What's weird actually is how ^9 works in browsers. It is *never* tab 9. It is always "last tab"!

**Comment by DHowett** [comment]
Well, it is tab 9 if the last tab is tab 9. I should not have said "never."

**Comment by zadjii-msft** [comment]
> If we're going down this road... For consistency's sake, I need ctrl+alt+9 to focus the last tab. That's how firefox, chromium, chrome, edge, _legacy edge_, and a bunch of other tabbed interfaces do it.

I mean, that's not too hard - setting it to 

```json
        { "command": { "action": "switchToTab", "index": 99999 }, "keys": "ctrl+alt+9" },
```

should work 😋

File: src/cascadia/TerminalSettingsModel/defaults.json [hunk 1]
```diff
@@ -329,7 +329,7 @@
         { "command": { "action": "switchToTab", "index": 5 }, "keys": "ctrl+alt+6" },
         { "command": { "action": "switchToTab", "index": 6 }, "keys": "ctrl+alt+7" },
         { "command": { "action": "switchToTab", "index": 7 }, "keys": "ctrl+alt+8" },
-        { "command": { "action": "switchToTab", "index": 8 }, "keys": "ctrl+alt+9" },
+        { "command": { "action": "switchToTab", "index": 99999 }, "keys": "ctrl+alt+9" },
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsModel/defaults.json:332]: will this, per chance, show up as...

image

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/defaults.json:332]: bah

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/defaults.json:332]: okay fine fine. I'll set it to `uint_whatever_t_max`, and special-case that resource as "switch to the last tab"

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsModel/defaults.json:332]: in latest commit

image

(I have the `"index":99999` version in `settings.json` to prove it works)

File: src/cascadia/LocalTests_TerminalApp/TabTests.cpp [hunk 2]
```diff
@@ -1342,4 +1344,55 @@ namespace TerminalAppLocalTests
         });
     }
 
+    void TabTests::TestClampSwitchToTab()
+    {
+        Log::Comment(L"Preview a color scheme. Make sure it's applied, then committed accordingly");
+
```
Comment (DHowett) [inline src/cascadia/LocalTests_TerminalApp/TabTests.cpp:1350]:
```suggestion
```

well not exactly, but we could also just remove the comment entirely

after all, taef logs the test name :smile:

**Review by DHowett — APPROVED** [review]

File: src/cascadia/LocalTests_TerminalApp/TabTests.cpp (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/LocalTests_TerminalApp/TabTests.cpp:1350]:
```suggestion
        Log::Comment(L"Test that switching to a tab index higher than the number of tabs just clamps to the last tab.");

```

File: src/cascadia/TerminalApp/TabManagement.cpp [hunk 3]
```diff
@@ -497,24 +497,25 @@ namespace winrt::TerminalApp::implementation
...
-                _UpdatedSelectedTab(tab);
-            }
-            else
-            {
-                _SetFocusedTab(tab);
-            }
+        // GH#9369 - if the argument is out of range, then clamp to the number
+        // of available tabs. Previously, we'd just silently do nothing if the
+        // value was greater than the number of tabs.
+        tabIndex = std::clamp(tabIndex, 0u, _tabs.Size() - 1);
```
Comment (lhecker) [inline src/cascadia/TerminalApp/TabManagement.cpp:505]: `tabIndex` can't be smaller than 0, so you can also use `std::min`.

**Review by lhecker — APPROVED** [review]

**Pull request merged**